rust/tokens_transfer/src/lib.rs

```diff
@@ -0,0 +1,103 @@
+use std::cell::RefCell;
+use std::collections::hash_map::DefaultHasher;
+use std::hash::{Hash, Hasher};
+use std::ops::Deref;
+use candid::{candid_method, CandidType};
+
+use ic_cdk::api::call::call;
+use ic_cdk_macros::*;
+use ic_ledger_types::{AccountIdentifier, BlockIndex, DEFAULT_SUBACCOUNT, MAINNET_LEDGER_CANISTER_ID, Memo, Subaccount, Tokens};
+use ic_types::Principal;
+use serde::{Deserialize, Serialize};
+
+#[cfg(test)] use std::path::PathBuf;
+#[cfg(test)] use std::io::Write;
+use ic_cdk::id;
+
+use crate::errors::TransferError;
+
+#[cfg(test)] mod candid_util;
+mod errors;
+
+#[derive(CandidType, Serialize, Deserialize, Clone, Debug, Hash, PartialEq)]
+pub struct Conf {
+    ledger_canister_id: Principal,
+    subaccount: Option<Subaccount>,
```

Comment:
Whose subaccount it is? Subaccount that this canister should always use?
I find it a bit weird: why would you fix a single subaccount? If you only allow one, just use the default one.

Reply:
I think it makes sense in the context of an example to show that there are several subaccounts for each account.

Reply:
I pushed a comment and add a small description and a link to the Ledger doc.